crates/languages/src/go/runnables.scm

```diff
@@ -1,9 +1,23 @@
+; Functions names start with `Test`
 (
-    (function_declaration name: (_) @run
-        (#match? @run "^Test.*"))
+    (function_declaration name: (_) @run @_name
+        (#match? @_name "^Test.+"))
 ) @go-test
 
+; `t.Run`
 (
-    (function_declaration name: (_) @run
-        (#eq? @run "main"))
-) @go-main
+    (call_expression function: (_) @run @_name
+        (#match? @_name "^t.Run.*"))
+) @go-subtest
```

Comment:
```suggestion
        (#match? @_name "^t.Run.*"))
        (#set! tag go-subtest)
) @go-subtest
```
You'll have to fix up the way runnables are tagged in this PR once https://github.com/zed-industries/zed/pull/12237 lands.